class/actions_subtotal.class.php

```diff
@@ -4168,20 +4198,39 @@ public function defineColumnField($parameters, &$pdfDoc, &$action, $hookmanager)
 	 */
 	private function _billOrdersAddCheckBoxForTitleBlocks()
 	{
-		global $delayedhtmlcontent, $langs;
+		global $delayedhtmlcontent, $langs, $conf;
+
 		ob_start();
+		$jsConf = array(
+			'langs'=>  array(
+				'AddTitleBlocFromOrdersToInvoice' => $langs->trans('subtotal_add_title_bloc_from_orderstoinvoice'),
+				'AddShippingListToTile' => $langs->trans('AddShippingListToTile'),
+				'SubtotalOptions' => $langs->trans('SubtotalOptions'),
+				'UseHiddenConfToAutoCheck' => $langs->trans('UseHiddenConfToAutoCheck'),
+			),
+			'isModShippingEnable' => !empty($conf->expedition->enabled),
+			'SUBTOTAL_DEFAULT_CHECK_SHIPPING_LIST_FOR_TITLE_DESC' => getDolGlobalInt('SUBTOTAL_DEFAULT_CHECK_SHIPPING_LIST_FOR_TITLE_DESC')
+		);
 		?>
 			<script type="text/javascript">
 				$(function() {
-					var tr = $("<tr><td><?php echo $langs->trans('subtotal_add_title_bloc_from_orderstoinvoice'); ?></td><td><input type='checkbox' value='1' name='subtotal_add_title_bloc_from_orderstoinvoice' checked='checked' /></td></tr>");
-					var $noteTextArea = $("textarea[name=note]");
+					let jsConf = <?php print json_encode($jsConf); ?>;
+
+					let tr = '<tr><td>'+jsConf.langs.SubtotalOptions+'</td><td>';
+					tr+= '<label><input type="checkbox" value="1" name="subtotal_add_title_bloc_from_orderstoinvoice" checked="checked" /> '+jsConf.langs.AddTitleBlocFromOrdersToInvoice+'</label>';
+					if(jsConf.isModShippingEnable){
+						tr+= '<br/><label><input type="checkbox" value="1" name="subtotal_add_shipping_list_to_title_desc" /> '+jsConf.langs.AddShippingListToTile+' <i class="fa fa-question-circle" title="'+jsConf.langs.UseHiddenConfToAutoCheck+' SUBTOTAL_DEFAULT_CHECK_SHIPPING_LIST_FOR_TITLE_DESC"></label>';
+					}
+					tr+= '<td></tr>';
```

Comment:
Pour la lisibilité (je sais pas si ça vaut vraiment le coup, mais je trouve ça plus simple à comprendre).
J'ai pas testé encore, mais en principe, ça ne doit rien changer.

```suggestion
					const checkboxWithLabel = (name, label, checked = false) => `<label><input type="checkbox" value="1" name="${name}" ${checked ? 'checked="checked"' : ''} />${label}</label>`;
					const buildInfoHelp = (helpText, faClass='fa-question-circle') => `<i class="fa ${faClass}" title="${helpText}"></i>`;
					let optCheckboxes = checkboxWithLabel('subtotal_add_title_bloc_from_orderstoinvoice', jsConf.langs.AddTitleBlocFromOrdersToInvoice, true);
					if (jsConf.isModShippingEnable) {
						optCheckboxes += checkboxWithLabel(
						        'subtotal_add_shipping_list_to_title_desc',
						        buildInfoHelp(jsConf.langs.UseHiddenConfToAutoCheck + ' SUBTOTAL_DEFAULT_CHECK_SHIPPING_LIST_FOR_TITLE_DESC')
						);
					}
					const tr = `<tr><td>${jsConf.langs.SubtotalOptions}</td><td>${optCheckboxes}</td></tr>`;
```

Reply:
Franchement je ne sais pas non plus, au début j'avais fait un peu comme toi mais après j'ai changé pour remettre un truc plus "à l'ancienne" car je me suis dit que ce serait plus compréhensible, mais je me suis peut-être fait des idées...
Du coup je ne sais pas. Mais vu que c'est toi maintenant le mainteneur du module, je dirais que c'est toi qui choisis ;-)